modules/edge-identity/src/identity.rs

```diff
@@ -69,43 +75,43 @@ pub struct IdentityRecord<AccountId, BlockNumber> {
 	pub stage: IdentityStage,
 	pub expiration_time: Option<BlockNumber>,
 	pub proof: Option<Attestation>,
-	pub verifications: Option<Vec<(AccountId, bool)>>,
 	pub metadata: Option<MetadataRecord>,
 }
 
 decl_module! {
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 		fn deposit_event<T>() = default;
 
-		/// Register an identity with the hash of the signature. Ensures that
-		/// all identities are unique, so that no duplicate identities can be
-		/// registered.
-		///
-		/// Current implementation suffers from squatter attacks. Additional
-		/// implementations could provide a mechanism for a trusted set of
-		/// authorities to delete a squatted identity OR implement storage
-		/// rent to disincentivize it.
-		pub fn register(origin, identity: Vec<u8>) -> Result {
+		pub fn register(origin, identity_type: Vec<u8>, identity: Vec<u8>) -> Result {
 			let _sender = ensure_signed(origin)?;
 			ensure!(!Self::frozen_accounts().iter().any(|i| i == &_sender.clone()), "Sender account is frozen");
-
-			let hash = T::Hashing::hash_of(&identity);
+			// Check that the sender hasn't used this identity type before
+			let mut buf = Vec::new();
+			buf.extend_from_slice(&_sender.encode());
+			buf.extend_from_slice(&identity_type.encode());
+			let type_hash = T::Hashing::hash(&buf[..]);
+			ensure!(!<UsedType<T>>::exists(type_hash), "Identity type already used");
+			<UsedType<T>>::insert(type_hash, true);
+			// Hash the identity type with the identity to use as a key for the mapping
+			buf = Vec::new();
+			buf.extend_from_slice(&identity_type.encode());
+			buf.extend_from_slice(&identity.encode());
+			let hash = T::Hashing::hash(&buf[..]);
```

Comment:
You could simplify this entire section by using a `map: AccountId => Array<IdentityType>` rather than tuple indexing.